helix-term/src/ui/completion.rs

```diff
@@ -79,7 +92,39 @@ impl menu::Item for CompletionItem {
                 Some(lsp::CompletionItemKind::ENUM) => "enum",
                 Some(lsp::CompletionItemKind::KEYWORD) => "keyword",
                 Some(lsp::CompletionItemKind::SNIPPET) => "snippet",
-                Some(lsp::CompletionItemKind::COLOR) => "color",
+                Some(lsp::CompletionItemKind::COLOR) => {
+                    let maybe_hex_color = match &item.documentation {
+                        Some(Documentation::String(text)) => Some(text),
+                        Some(Documentation::MarkupContent(MarkupContent { value, .. })) => {
+                            Some(value)
+                        }
+                        None => None,
+                    };
+
+                    let content = maybe_hex_color.map_or(Span::raw("color"), |hex_color| {
+                        let maybe_color = match (
+                            hex_color
+                                .get(1..=2)
+                                .and_then(|r| u8::from_str_radix(r, 16).ok()),
+                            hex_color
+                                .get(3..=4)
+                                .and_then(|g| u8::from_str_radix(g, 16).ok()),
+                            hex_color
+                                .get(5..=6)
+                                .and_then(|b| u8::from_str_radix(b, 16).ok()),
+                        ) {
+                            (Some(r), Some(g), Some(b)) => Some(Color::Rgb(r, g, b)),
+                            _ => None,
+                        };
+
+                        match maybe_color {
+                            Some(color) => Span::styled("        ", Style::default().bg(color)),
+                            None => Span::raw("color"),
+                        }
+                    });
+
+                    return menu::Row::new([first_cell, menu::Cell::from(content)]);
```

Comment:
I prefered the `.into()` from earlier in the PR. Returning here makes the change smaller but this block becomes harder to follow and we duplicate the definition of the row

Reply:
refactored in https://github.com/helix-editor/helix/pull/12299/commits/746b7a5f6f69914cdc7077906753a60ead79b79c